File: archivo-historico/ejemplos/nerve/02-pub-sub-basico/publicador.go
package main

import (
	"fmt"
	"os"
	"os/signal"
	"syscall"
	"time"

	"nerve"
)

func main() {
	fmt.Println("Iniciando publicador (estación de radio)...")
	cliente := nerve.NewNexusClient()
	cliente.Connect("publicador_noticias")
	fmt.Println("✅ Publicador conectado al Hub.")

	ticker := time.NewTicker(3 * time.Second)
	quit := make(chan os.Signal, 1)
	signal.Notify(quit, os.Interrupt, syscall.SIGTERM)

	contador := 1
	
	go func() {
		for {
			select {
			case <-ticker.C:
				noticia := map[string]interface{}{
					"canal":     "noticias_tech",
					"titular":   fmt.Sprintf("Noticia #%d: Nerve Hub funciona genial", contador),
					"timestamp": time.Now().Unix(),
				}

				fmt.Printf(" Transmitiendo (broadcast): %s\n", noticia["titular"])
				cliente.Broadcast(noticia)
				contador++
			}
		}
	}()

	<-quit
	fmt.Println("\nApagando publicador...")
	ticker.Stop()
	cliente.Disconnect()
}

File: archivo-historico/ejemplos/nerve/02-pub-sub-basico/publicador.py
import time
from nerve import NexusClient


def main():
    print("Iniciando publicador (estación de radio)...")
    cliente = NexusClient()
    cliente.connect("publicador_noticias")
    print("✅ Publicador conectado al Hub.")

    contador = 1
    try:
        while True:
            noticia = {
                "canal": "noticias_tech",
                "titular": f"Noticia #{contador}: Nerve Hub funciona genial",
                "timestamp": time.time(),
            }

            # Usamos broadcast para enviarlo a todos los nodos conectados
            print(f" Transmitiendo (broadcast): {noticia['titular']}")
            cliente.broadcast(noticia)

            contador += 1
            time.sleep(3)  # Transmite cada 3 segundos

    except KeyboardInterrupt:
        print("\nApagando publicador...")
        cliente.disconnect()


if __name__ == "__main__":
    main()
